Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -345,6 +346,20 @@ - (void)setExternalIntentURI:(NSString *)externalIntentURI {
     }
 }
 
+- (NSString*) referredUrl {
+    @synchronized (self) {
+        if (!_referredUrl) _referredUrl = [self readStringFromDefaults:@"referredUrl"];
```

Comment:
The rest of this file uses constants with a specific pattern for the keys to ns user defaults - why are you deviating here?